src/microsim/schema/modality/_simple.py

```diff
@@ -39,22 +43,39 @@ def render(
         settings: Settings,
         xp: NumpyAPI | None = None,
     ) -> xrDataArray:
-        if truth.ndim > 4 or Axis.F not in truth.dims:
-            raise NotImplementedError(
-                "At this stage, we only support rendering 3D or 4D data with a "
-                "fluorophore dimension."
-            )
+        convolved = 0
+        for fluor_idx in range(truth.sizes[Axis.F]):
+            convolved_fluor = 0
+            for bin_idx in range(truth.sizes[Axis.W]):
+                binned_flux = truth.isel({Axis.W: bin_idx, Axis.F: fluor_idx})
+                if np.isnan(np.sum(binned_flux.data)):
+                    # NOTE: there can be bins for which there is no data in one of the
+                    #  fluorophores
+                    continue
+                em_wvl = binned_flux[Axis.W].values.item().mid * UnitRegistry().nm
```

Comment:
it's important (and sneaky) in pint that you use the same unitRegistry everywhere. here you're instantiating a new one (not accessing the previous global one).  See the top of `emission.py` for a pattern that you can use to access the same global ureg everywhere

```python
ureg = pint.application_registry.get()
```